File: ublockfilterutility.client/src/app/shared/models/filter.model.ts
import { FormControl } from "@angular/forms";
import { FilterParameters } from "./filter-parameter.model";

export interface FilterModel {
    Id: number;
    Name: string;
    Parameters: FilterParameters;
    Template: string;
}

export interface ApiFilterRequest {
    id?: string;
    filter: FilterModel;
}

File: uBlockFilterUtility.Server/Models/FilterModel.cs
namespace uBlockFilterUtility.Models
{
    public class FilterModel
    {
        public int? Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public Dictionary<string, string>? Parameters { get; set; } = new Dictionary<string, string>();

        public string Template { get; set; } = string.Empty;
    }
}
